### PR TITLE
Fix front of 2 sided doc upload preview visible in PoA step

### PR DESCRIPTION
**Pull request by kopijunkie**
# Problem
Some clients want to set up the SDK with PoA step ordered after Document Capture step.

If a user selects a 2-sided ID Document after both sides are uploaded and the user moves on to the PoA step, after the PoA Document Select view they are taken straight to the PoA Confirm view but with the front side of the earlier uploaded document instead. If they were to click on the Back button, this would go back to the PoA Capture view they were originally expecting.

# Solution
Keep ID document type separate from PoA document type by adding a `poaDocumentType` to the store's state along with a corresponding `setPoaDocumentType` action rather than re-using `documentType` to keep track of both ID and PoA Document Type user selection

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [ ] Have new automated tests been implemented?
- [ ] Have new manual tests been written down?
- [x] Have tests passed locally?
- [n/a] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://773-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Router/index.js [hunk 1]
```diff
@@ -92,8 +92,17 @@ class CrossDeviceMobileRouter extends Component {
...
   }
 
-  setConfig = (actions) => (data) => {
-    const {token, steps, language, documentType, step: userStepIndex,clientStepIndex, woopraCookie} = data
+  setMobileConfig = (actions) => (data) => {
+    const {
+      token,
+      steps,
+      language,
+      documentType,
```
Comment (rfreitas) [inline src/components/Router/index.js]: it will be interesting to double check with @joninvski to make sure backend wise we are good with this change.

Comment (kopijunkie) [reply src/components/Router/index.js]: Yes, good point. Will check with him if this will play nice with backend.

Comment (joninvski) [reply src/components/Router/index.js]: Just checked it. We are ok. We just blacklist some fields:

https://gitlab.eu-west-1.mgmt.onfido.xyz/onfido/sdks/cross_device_sync/blob/3279fb2c90cbc62fe66033bb64def40786c0c680/src/message_handle.js#L113

And none of the blacklisted names are there:

https://gitlab.eu-west-1.mgmt.onfido.xyz/onfido/sdks/cross_device_sync/blob/3279fb2c90cbc62fe66033bb64def40786c0c680/src/message_handle.js#L17

So :+1: from the backend side

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by seewah** [comment]
the fix seems ok. I have tested on all browser. One thing i have spotted is that we are now sending type = `unknown` to API when submitting poa documents. I would also expect to see the correct type in the callback too.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-773-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://773-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**